### PR TITLE
pod probe marker proposal

### PR DESCRIPTION
**Pull request by zmberg**
Signed-off-by: liheng.zms <liheng.zms@alibaba-inc.com>
### Ⅰ. Describe what this PR does
1. pod probe marker proposal

### CODE REVIEW

**Comment by pigletfly** [comment]
what's the relation between PodProbeMaker probes and the default kubernetes probes?If both set, both will executed?

File: docs/proposals/20220728-pod-probe-marker.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      Exec: /home/game/idle.sh
+      # HttpGet: xxxx
+      # TcpSocket: xxxx
+    - Name: Master
+      Exec: /home/game/master.sh
+  markItems:
+    # When probe execution is successful
+  - probeName: Master
+    expectation: succeeded
+    labels:
```
Comment (furykerry) [inline docs/proposals/20220728-pod-probe-marker.md:69]: plz enumerate possible actions, i think the following action is possible
1. patch labels
2. patch annotations
3. patch readiness gate conditions

File: docs/proposals/20220728-pod-probe-marker.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+kind: PodProbeMarkerRun
+metadata:
+  name: pod-name
+spec:
+  containers:
+  - gameserver
+    probes
+    - Name: Idle
+      Exec: xxxx
+      HttpGet: xxxx
```
Comment (furykerry) [inline docs/proposals/20220728-pod-probe-marker.md:87]: plz enumerate necessary probe settings such as initialdelay, and plz comments the relationship with startup probe

File: docs/proposals/20220728-pod-probe-marker.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    # When probe execution is successful
+  - probeName: Healthy
+    expectation: succeeded
+    labels:
+      gameserver-healthy: 'true'
+  - probeName: Healthy
+    expectation: failed
+    labels:
+      gameserver-healthy: 'false'
+  - probeName: Idle
```
Comment (furykerry) [inline docs/proposals/20220728-pod-probe-marker.md:74]: consider more concise form of the probe spec

```
  - probeName: Idle
     - condition: succeeded
        annotations:
            controller.kubernetes.io/pod-deletion-cost: '-1'
     - condition: failed
        annotations:
            controller.kubernetes.io/pod-deletion-cost: '1'
```

File: docs/proposals/20220728-pod-probe-marker.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      tcpSocket:
+        port: 8899
+  markerItems:
+    # When probe execution is successful
+  - probeName: Healthy
+    conditions:
+    - status: 'True'
+      labels:
+        gameserver-healthy: 'true'
+    - status: 'False'
```
Comment (furykerry) [inline docs/proposals/20220728-pod-probe-marker.md:71]: how we ensure there conditions True and False both exists, what if one of them(such as False) is not specified, what is the default behavior, clear the annotations?

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**